### PR TITLE
remote-check-token-invalid

### PR DESCRIPTION
**Pull request by Clement-Leopold**
issue #1355 Getting invalid token using spring boot autoconfigure for the resource server 

when I try to use the default RemoteTokenServices to check whether my token is valid in Spring Cloud, it failed. 
And When I debug line by line,  I find the line 116 in RemoteTokenServices checking the 'active' whether that is true or not, it failed to tell when it is true.  
So I override the RemoteTokenServices, and changed the line to be this, it succeeded.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Clement-Leopold Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1440)!

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
 		}
 
 		// gh-838
-		if (!Boolean.TRUE.equals(map.get("active"))) {
+		if (!Boolean.TRUE.toString().equals(map.get("active"))) {
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java]: @Clement-Leopold With this change, `"true"` would work, but `true` would not. The provider may return the `active` attribute as either a JSON boolean or string. We need to account for both types.

Can you please update and add a test as well?

**Comment by Clement-Leopold** [comment]
I've added one test and modified the condition code, if you have any suggestions please let me know, thanks!

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java (hunk 1 above)
Comment (Clement-Leopold) [reply spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java]: @jgrandja  I've made some updates as you requested. thanks!

**Comment by Clement-Leopold** [comment]
@jgrandja  I am so sorry for that once I've pushed the author name.  This is my fault.

**Comment by jgrandja** [comment]
@Clement-Leopold I haven't had a chance to review this as of yet. We've been very busy working towards the upcoming Spring Security 5.1 release. I'm planning on circling back to this within the next couple of weeks. Thanks for your patience.

**Comment by jgrandja** [comment]
@Clement-Leopold I'm going to close this PR as it's been addresses (and merged) in #1533

**Pull request closed without merging**